packages/graphql-client/src/graphql-client/tests/graphql-client/client-requestStream.test.ts

```diff
@@ -0,0 +1,1457 @@
+                    const multipleResponsesArray = [
+                      `
+                        --graphql
+                        Content-Type: application/json
+                        Content-Length: 120\r\n\r\n{"data":{"shop":{"id":"${id}"}},"extensions":{"context": {"country": "${variables.country}", "language": "${variables.language}"}},"hasNext":true}
+                        --graphql
+
+                        `,
+                    ];
+
+                    const mockedSuccessResponse = responseGenerator(
+                      multipleResponsesArray,
+                    );
+
+                    fetchMock.mockResolvedValue(mockedSuccessResponse);
+
+                    const responseStream = await client.requestStream(
+                      operation,
+                      {
+                        variables,
+                      },
+                    );
+
+                    const results: any = [];
+
+                    for await (const response of responseStream) {
+                      results.push(response);
+                    }
+
+                    const lastResponse = results.slice(-1)[0];
+                    expect(lastResponse.data).toEqual({
+                      shop: { id },
+                    });
+
+                    expect(lastResponse.errors).toEqual({
+                      networkStatusCode: 200,
+                      message:
+                        "GraphQL Client: Response stream terminated unexpectedly",
+                      response: mockedSuccessResponse,
+                    });
+                  });
+                });
+
+                it("returns an async iterator that returns a response object with no data value and a JSON parsing error if the returned data is a malformed JSON", async () => {
+                  const multipleResponsesArray = [
+                    `
+                      --graphql
+                      Content-Type: application/json
+                      Content-Length: 120\r\n\r\n{"data":{"shop":{"id":"${id}"}}},"extensions":{"context": {"country": "${variables.country}", "language": "${variables.language}"}},"hasNext":false}
+                      --graphql--
+                      `,
+                  ];
+
+                  const mockedSuccessResponse = responseGenerator(
+                    multipleResponsesArray,
+                  );
+
+                  fetchMock.mockResolvedValue(mockedSuccessResponse);
+
+                  const responseStream = await client.requestStream(operation, {
+                    variables,
+                  });
+
+                  const results: any = [];
+
+                  for await (const response of responseStream) {
+                    results.push(response);
+                  }
+
+                  const response = results[0];
+                  const errors = response.errors;
+                  expect(errors.networkStatusCode).toBe(200);
+                  expect(errors.message).toMatch(
```

Comment:
Since errors can happen in between chunks within the stream, the response itself might not be very useful since there might not be any relevant info. 

Let's see if the community has feedback on this and if so, we can re-address it later.